clouddriver-alicloud/src/main/java/com/netflix/spinnaker/clouddriver/alicloud/provider/agent/AliCloudLoadBalancerCachingAgent.java

```diff
@@ -0,0 +1,408 @@
+import com.netflix.spinnaker.cats.agent.CachingAgent;
+import com.netflix.spinnaker.cats.agent.DefaultCacheResult;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.cats.cache.DefaultCacheData;
+import com.netflix.spinnaker.cats.provider.ProviderCache;
+import com.netflix.spinnaker.clouddriver.alicloud.AliCloudProvider;
+import com.netflix.spinnaker.clouddriver.alicloud.cache.Keys;
+import com.netflix.spinnaker.clouddriver.alicloud.provider.AliProvider;
+import com.netflix.spinnaker.clouddriver.alicloud.security.AliCloudClientProvider;
+import com.netflix.spinnaker.clouddriver.alicloud.security.AliCloudCredentials;
+import com.netflix.spinnaker.clouddriver.alicloud.security.AliCloudCredentialsProvider;
+import com.netflix.spinnaker.clouddriver.cache.OnDemandAgent;
+import com.netflix.spinnaker.clouddriver.cache.OnDemandMetricsSupport;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+
+public class AliCloudLoadBalancerCachingAgent implements CachingAgent, AccountAware, OnDemandAgent {
+  private AliProvider aliProvider;
+  private AliCloudClientProvider aliCloudClientProvider;
+  private AliCloudCredentials account;
+  private String region;
+  private AliCloudCredentialsProvider aliCloudCredentialsProvider;
+  ObjectMapper objectMapper;
+  OnDemandMetricsSupport metricsSupport;
+  IAcsClient client;
+
+  public AliCloudLoadBalancerCachingAgent(
+      AliProvider aliProvider,
+      String region,
+      AliCloudClientProvider aliCloudClientProvider,
+      AliCloudCredentialsProvider aliCloudCredentialsProvider,
+      AliCloudProvider aliCloudProvider,
+      ObjectMapper objectMapper,
+      Registry registry,
+      AliCloudCredentials credentials,
+      IAcsClient client) {
+    this.account = credentials;
+    this.aliProvider = aliProvider;
+    this.region = region;
+    this.aliCloudClientProvider = aliCloudClientProvider;
+    this.aliCloudCredentialsProvider = aliCloudCredentialsProvider;
+    this.objectMapper = objectMapper;
+    this.metricsSupport =
+        new OnDemandMetricsSupport(
+            registry,
+            this,
+            aliCloudProvider.getId()
+                + ":"
+                + aliCloudProvider.getId()
+                + ":"
+                + OnDemandAgent.OnDemandType.LoadBalancer);
+    this.client = client;
+  }
+
+  @Override
+  public String getAccountName() {
+    return null;
+  }
+
+  @Override
+  public Collection<AgentDataType> getProvidedDataTypes() {
+    return new ArrayList<AgentDataType>() {
+      {
+        add(AUTHORITATIVE.forType(LOAD_BALANCERS.ns));
+        add(INFORMATIVE.forType(INSTANCES.ns));
+      }
+    };
+  }
+
+  @Override
+  public CacheResult loadData(ProviderCache providerCache) {
+    logger.info(this.getClass().getSimpleName() + " is working 2");
```

Comment:
The class name is going to be included as part of the logger output. That should be removed.